File: archive/factorial.go
package main

import "fmt"

func factorial(n int) int {
        if n <= 1 {
                return 1
        }

        return n * factorial(n - 1)
}

func main() {
        var x int
        fmt.Println("Enter a number to find the factorial of:")
        _, err = fmt.Scan(&x)
	if err != nil {
		panic(err)
	}
        x = factorial(x)

        fmt.Println(x)
}

File: archive/factorial.rs
use std::io;

fn factorial(num: u32) -> u32 {
    match num {
        0 => 1,
        1 => 1,
        _ => factorial(num - 1) * num,
    }
}

fn main() {
    let mut number = String::new();
    io::stdin()
        .read_line(&mut number)
        .expect("Readline failure");

    let number: u32 = match number.trim().parse() {
        Ok(num) => num,
        Err(_) => return,
    };

    println!("Enter a number to find the factorial of: ");
    println!("{}", factorial(number));
}
